addons/web/static/src/core/py_js/py_date.js

```diff
@@ -500,3 +500,153 @@ export class PyRelativeDelta {
         this.second = 0;
     }
 }
+
+const TIME_DELTA_KEYS = "weeks days hours minutes seconds milliseconds microseconds".split(" ");
+
+/**
+ * Passes the fractional and integer parts of x to the callback, returns
+ * the callback's result
+ */
+function modf(x, fn) {
```

Comment:
why not use a synchronous api? it's just a pure function
```js
function modf(x) {
  const mod = x % 1;
  return [mod < 0 ? mod + 1 : mod, Math.floor(x)];
```

// can be used like this:
let [value, floor] = modf(x);

Reply:
by the way, maybe we should do that with divmod as well. it makes it so complicated to have a callback function

Reply:
not done in the commit. Maybe after that?

Reply:
> why not use a synchronous api? it's just a pure function
> 
> ```js
> function modf(x) {
>   const mod = x % 1;
>   return [mod < 0 ? mod + 1 : mod, Math.floor(x)];
> }
> 
> // can be used like this:
> let [value, floor] = modf(x);
> ```

donde